### DISCUSSION TITLE
Handling Multiple snack bars Display Issues in Concurrent API Calls.

### DISCUSSION THREAD

**Discussion by khushpajwani** [jonataslaw/getx#3151] [Q&A] [answered]
I am encountering an issue with displaying multiple snack bars in quick succession. For example, when I initiate five API calls simultaneously without waiting for their responses, and if any of these APIs fail, I attempt to show snack bars to indicate the errors. However, when multiple snack bars are triggered in rapid succession, the application crashes.

**Comment by dharambudh1** [comment, marked as answer]
Try using this code:

```
  void showSnackbar({required String title, required String message}) {
    unawaited(
      Get.closeCurrentSnackbar().whenComplete(
        () {
          Get.snackbar(title, message, instantInit: false);
          return;
        },
      ),
    );
    return;
  }
```

Explanation:
The `showSnackbar` function first attempts to close any currently displayed snack bar using `Get.closeCurrentSnackbar()`. Once the current snack bar is closed (as ensured by the `whenComplete` callback), it then displays a new snack bar with the provided title and message. The use of `unawaited` suggests that the function is designed to run independently of the completion of the `Future` from `Get.closeCurrentSnackbar()`, meaning that it does not block or wait for the snack bar to close before executing the next lines of code.

**Comment by khushpajwani** [comment]
Okay, thank you!
